index.js

```diff
@@ -80,16 +98,20 @@ function ensureProperties (obj) {
     try {
       for (var prop in obj) {
         if (!Object.prototype.hasOwnProperty.call(obj, prop)) continue
+        if (isDescendent(filterPaths, path.join('.')) && shouldFilter(filterKeys, prop)) {
```

Comment:
is there a reason we are using `join` here instead of having path be a string and doing concatenation as we go?

Reply:
I was considering replacing the `depth` variable with `path.length` seeing as how without giving it much thought it would be equivalent. But I got concerned about making that change and so didn't end up doing it.